src/index.ts

```diff
@@ -0,0 +1,518 @@
+   */
+  public subscribe(handler: Function): Unsubscribe {
+    this.subscribers.push(handler);
+
+    return () => {
+      const index: number = this.subscribers.indexOf(handler);
+      this.subscribers.splice(index, 1);
+    };
+  }
+
+  /**
+   * Find an appropriate string for a given key. This function will attempt to find the string
+   * using the primary locale, then will use the fallback if there is no match. If there are no
+   * matches, then the key will be returned.
+   *
+   * @param key   The key for looking up the string. Context should be delimited with a period.
+   * @param data  Contains data to be interpolated. `count` is a magic value for pluralization.
+   * @returns     Best string match for key given.
+   */
+  public t(key: string, data: TDataObject = {}, options: TOptions = {}): string {
+    // If we're not in production mode, be sure the people testing
+    if (process.env.NODE_ENV !== 'production') {
+      if (!key.match(VALID_KEY)) {
+        throw new Error(
+          `${key} is not a valid key format. ` +
+          `Valid keys are of the format ${VALID_KEY.toString()}`);
+      }
+    }
+
+    // If key mode is enabled, simply return the key.
+    if (this.getKeyMode()) {
+      return key;
+    }
+
+    const locales: string[] = [];
+
+    // If options has a locale, normalize it and use that instead of locale. Otherwise, check
+    // if locale is even set and add that to possible locales.
+    if (options.locale !== undefined) {
+      locales.push(this.normalizeLocale(options.locale));
+    } else if (this.locale !== undefined) {
+      locales.push(this.locale);
+    }
+
+    if (options.fallback !== undefined) {
+      locales.push(this.normalizeLocale(options.fallback));
+    } else {
+      locales.push(this.fallback);
+    }
+
+    // Loop through the locales and try to find an appropriate translation.
+    for (let i = 0; i < locales.length; i++) {
+      const locale = locales[i];
+      const contexts = key.split('.');
+
+      let translation: TranslationLookup =
+        contexts.reduce(this.translationReducer, this.translations[locale]);
+
+      // If there is a count and an object comes back, check if this key is a plural context.
+      if (typeof translation === 'object' && typeof data.count === 'number') {
+        const pluralContext: string = this.getPluralContext(locale, data.count);
+        translation = translation[pluralContext];
+      }
+
+      // If the translation is equal to string, interpolate and return.
+      if (typeof translation === 'string') {
+        return translation
+          .replace(INTERPOLATION_REGEX, (exp, arg) => {
+            const result = data[arg];
+            return typeof result === 'string' ? result :
+                   typeof result === 'number' ? `${result}` :
+                   exp;
+          });
+      }
+    }
+
+    // Fallback to the key.
+    return key;
+  }
+
+  protected callSubscribers(): void {
```

Comment:
Protected for extensibility purposes. I didn't want to build our time locale methods into the library because I would rather encourage them to use `Intl` over `moment`. I also wanted to avoid trying to parse timezone data and adding that as a dependency. Although, it might make sense to do this in the future.